server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -872,6 +873,17 @@ public void writeVIntArray(int[] values) throws IOException {
         }
     }
 
+    public void writeBigIntArray(IntArray values) throws IOException {
+        // TODO do we need to specialize this for ReleasableIntArray
+        if (values.size() > Integer.MAX_VALUE / Integer.BYTES) {
+            throw new UnsupportedOperationException(); // NOCOMMIT What do?
+        }
```

Comment:
We rarely store anything close to this many ints in aggs land. We're not going to do this. That's 2gb of data. 536 millions buckets worth of integers. 268 million buckets of longs or doubles. I suppose one day we'd love to handle this, but I can't imagine our strategy of slamming all of them over the wire at once is really compatible with that now. I'd saw we're fairly safe with a limit like this.